main.tf

```diff
@@ -1,4 +1,40 @@
-resource "google_storage_bucket" "example" {
-  name     = var.name
-  location = "EU"
+data "aws_caller_identity" "current" {}
+
+# Create a secret in AWS Secrets Manager to store the upstream registry name.
+resource "aws_secretsmanager_secret" "ecr_pullthroughcache" {
+  name = "ecr-pullthroughcache/${var.upstream_registry}"
+}
+
+resource "aws_secretsmanager_secret_version" "ecr_pullthroughcache" {
+  secret_id     = aws_secretsmanager_secret.ecr_pullthroughcache.id
+  secret_string = jsonencode(var.credentials)
```

Comment:
As detailed in the comment in the variable, I would remove the possibility for the user/project to specify its credentials in a variable. The `ignore_changes` is another thing that we have to consider if the credentials can be configured here.